sklearn/decomposition/_dict_learning.py

```diff
@@ -1469,6 +1469,7 @@ class DictionaryLearning(_BaseSparseCoding, BaseEstimator):
     ...     n_samples=100, n_components=15, n_features=20, n_nonzero_coefs=10,
     ...     random_state=42,
     ... )
+    >>> X, dictionary, code = X.T, dictionary.T, code.T
     >>> dict_learner = DictionaryLearning(
     ...     n_components=15, transform_algorithm='lasso_lars', random_state=42,
     ... )
```

Comment:
With these params, the penalization is too important and X_transformed ends up having all zeros for some samples. I suggest to change to the following
```suggestion
    >>> dict_learner = DictionaryLearning(
    ...     n_components=15, transform_algorithm='lasso_lars', transform_alpha=0.1,
    ...     random_state=42,
    ... )
```